### ISSUE TITLE
Fetch LABELs once before program executes

### ISSUE THREAD

**Issue by worksofliam** [worksofbarry/NetRPG#25]
Every time we execute a procedure we are building the jump tables.

This can be seen here: https://github.com/worksofbarry/NetRPG/blob/master/NetRPG/Runtime/VM.cs#L187

This is only really noticeable when you are making use of procedures.

We can really make it better if we just build this table once per procedure.

**Comment by worksofliam** [comment]
Length of tests before the change (`mm:ss:f:ff`):

```
Testing dcl_char.rpgle                      ... successful (00:00:2:23).
Testing dcl_int.rpgle                       ... successful (00:00:0:09).
Testing dcl_fixed.rpgle                     ... successful (00:00:0:05).
Testing dcl_ind.rpgle                       ... successful (00:00:1:10).
Testing dcl_char_array.rpgle                ... successful (00:00:0:05).
Testing dcl_ds_qualified.rpgle              ... successful (00:00:0:09).
Testing dcl_ds_qualified_array.rpgle        ... successful (00:00:1:10).
Testing dcl_ds_subf_array.rpgle             ... successful (00:00:0:06).
Testing dcl_ds_nonqualified.rpgle           ... successful (00:00:0:05).
Testing dcl_array_subf_array.rpgle          ... successful (00:00:0:07).
Testing dcl_proc_1.rpgle                    ... successful (00:00:0:03).
Testing dcl_proc_2.rpgle                    ... successful (00:00:0:04).
Testing dcl_proc_3.rpgle                    ... successful (00:00:0:05).
Testing dcl_proc_4.rpgle                    ... successful (00:00:0:08).
Testing dcl_proc_5.rpgle                    ... successful (00:00:0:05).
Testing dcl_proc_6.rpgle                    ... successful (00:00:2:20).
Testing dcl_proc_7.rpgle                    ... successful (00:00:0:05).
Testing dcl_proc_8.rpgle                    ... successful (00:00:0:07).
Testing dcl_proc_9_value.rpgle              ...
Before: world
After: world!
Before: hello
After: hello!
successful (00:00:1:10).
Testing dcl_proc_10_value.rpgle             ... successful (00:00:0:05).
Testing dcl_inz.rpgle                       ... successful (00:00:1:12).
Testing dcl_likeds.rpgle                    ... successful (00:00:0:05).
Testing assignment.rpgle                    ... successful (00:00:0:06).
Testing assignment2.rpgle                   ... successful (00:00:0:04).
Testing assignment3.rpgle                   ... successful (00:00:0:06).
Testing addition.rpgle                      ... successful (00:00:0:08).
Testing subtraction.rpgle                   ... successful (00:00:0:05).
Testing division.rpgle                      ... successful (00:00:0:05).
Testing multiplication.rpgle                ... successful (00:00:0:05).
Testing bif_abs.rpgle                       ... successful (00:00:0:00).
Testing bif_char.rpgle                      ... successful (00:00:0:01).
Testing bif_dec.rpgle                       ... successful (00:00:0:02).
Testing bif_decpos.rpgle                    ... successful (00:00:0:02).
Testing bif_float.rpgle                     ... successful (00:00:0:02).
Testing bif_int.rpgle                       ... successful (00:00:0:02).
Testing bif_len.rpgle                       ... successful (00:00:0:01).
Testing bif_lookup.rpgle                    ... successful (00:00:0:04).
Testing bif_trim.rpgle                      ... successful (00:00:0:01).
Testing bif_trim_2.rpgle                    ... successful (00:00:0:05).
Testing bif_trimr.rpgle                     ... successful (00:00:0:01).
Testing bif_triml.rpgle                     ... successful (00:00:0:01).
Testing bif_elem1.rpgle                     ... successful (00:00:0:03).
Testing bif_elem2.rpgle                     ... successful (00:00:0:01).
Testing bif_scan.rpgle                      ... successful (00:00:0:05).
Testing bif_scanrpl.rpgle                   ... successful (00:00:0:02).
Testing bif_xlate1.rpgle                    ... successful (00:00:0:04).
Testing bif_xlate2.rpgle                    ... successful (00:00:0:05).
Testing bif_subst.rpgle                     ... successful (00:00:0:09).
Testing bif_subst2.rpgle                    ... successful (00:00:0:02).
Testing bif_replace.rpgle                   ... successful (00:00:0:06).
Testing time_dcl_timestamp.rpgle            ... successful (00:00:0:01).
Testing time_bif_timestamp.rpgle            ... successful (00:00:0:02).
Testing time_bif_timestamp2.rpgle           ... successful (00:00:0:02).
Testing time_bif_timestamp3.rpgle           ... successful (00:00:0:02).
Testing time_bif_date1.rpgle                ... successful (00:00:0:02).
Testing time_bif_time1.rpgle                ...
01:23 PM
48180
48180
successful (00:00:0:04).
Testing time_bif_days.rpgle                 ... successful (00:00:0:02).
Testing time_bif_hours.rpgle                ... successful (00:00:0:01).
Testing time_bif_minutes.rpgle              ... successful (00:00:0:01).
Testing time_bif_seconds.rpgle              ... successful (00:00:0:01).
Testing time_date_literal.rpgle             ... successful (00:00:0:04).
Testing time_time_literal.rpgle             ... successful (00:00:0:04).
Testing time_dcl_date.rpgle                 ... successful (00:00:0:01).
Testing time_dcl_time.rpgle                 ... successful (00:00:0:02).
Testing time_bif_char1.rpgle                ... successful (00:00:0:02).
Testing time_bif_char2.rpgle                ... successful (00:00:0:02).
Testing time_bif_char3.rpgle                ... successful (00:00:0:02).
Testing time_bif_char4.rpgle                ...
860198400
successful (00:00:0:04).
Testing time_bif_diff_days.rpgle            ... successful (00:00:0:04).
Testing time_bif_diff_seconds.rpgle         ... successful (00:00:0:04).
Testing time_bif_subdt.rpgle                ... successful (00:00:0:07).
Testing op_if.rpgle                         ... successful (00:00:0:08).
Testing op_else.rpgle                       ... successful (00:00:0:03).
Testing op_elseif.rpgle                     ... successful (00:00:0:05).
Testing op_select.rpgle                     ... successful (00:00:0:09).
Testing op_other1.rpgle                     ... successful (00:00:0:08).
Testing op_other2.rpgle                     ... successful (00:00:1:10).
Testing op_dow.rpgle                        ... successful (00:00:0:04).
Testing op_dow2.rpgle                       ... MyInt is 5
successful (00:00:0:06).
Testing op_dsply.rpgle                      ... Hello world
successful (00:00:0:00).
Testing op_eval1.rpgle                      ... successful (00:00:0:04).
Testing op_reset1.rpgle                     ... successful (00:00:0:02).
Testing op_reset2.rpgle                     ... successful (00:00:0:02).
Testing op_reset3.rpgle                     ... successful (00:00:0:07).
Testing op_reset4.rpgle                     ... successful (00:00:0:05).
Testing op_reset5.rpgle                     ... successful (00:00:1:10).
Testing op_clear1.rpgle                     ... successful (00:00:0:02).
Testing op_clear2.rpgle                     ... successful (00:00:0:02).
Testing op_clear3.rpgle                     ... successful (00:00:0:06).
Testing op_clear4.rpgle                     ... successful (00:00:0:05).
Testing op_in1.rpgle                        ... successful (00:00:0:01).
Testing op_in2.rpgle                        ... successful (00:00:0:01).
Testing op_in3.rpgle                        ... successful (00:00:0:02).
Testing op_file_open.rpgle                  ... successful (00:00:2:26).
Testing op_file_read.rpgle                  ... successful (00:00:0:01).
Testing op_file_read2.rpgle                 ... successful (00:00:0:01).
Testing op_file_read_qualified.rpgle        ... successful (00:00:0:02).
Testing op_file_readp.rpgle                 ... successful (00:00:0:02).
Testing op_file_chain.rpgle                 ... successful (00:00:0:02).
Testing op_file_chain2.rpgle                ... successful (00:00:0:05).
Testing op_file_chain3.rpgle                ... successful (00:00:0:03).
Testing op_file_write.rpgle                 ...
My first p
My new pro

successful (00:00:0:04).
Testing bif_eof1.rpgle                      ...
My first p
My second
successful (00:00:0:04).
Testing bif_eof2.rpgle                      ... successful (00:00:0:04).
Testing ind1.rpgle                          ... successful (00:00:0:01).
Testing ind2.rpgle                          ... successful (00:00:0:01).
Testing ind3.rpgle                          ...
LR before = 0
LR after  = 1
successful (00:00:0:03).
Testing dcl_shared1.rpgle                   ... Hi
Hi
successful (00:00:0:05).
Testing ile_module_a.rpgle                  ... Testing ile_module_b.rpgle                  ...
You are Liam, you are 22 years old

successful (00:00:1:17).
Testing system_printf.rpgle                 ... Hello world!
successful (00:00:0:07).
Testing system_dq.rpgle                     ... Before:
After: Hello world
successful (00:00:2:21).

111 ran, 111 passed, 0 failed
```

**Comment by worksofliam** [comment]
Implemented in 60b5f2cce8444e15e828a49ebf79dd25b88c50dd
